### PR TITLE
make a prefix to the copied cart rule ('copy of XXX')

### PR DESCRIPTION
**Pull request by thyseus**
and properly save related channels and customer groups after replication

### CODE REVIEW

File: packages/Webkul/CartRule/src/Http/Controllers/CartRuleController.php [hunk 1]
```diff
@@ -85,10 +85,21 @@ public function copy(int $cartRuleId): View
...
             ->replicate()
-            ->fill(['status' => 0]);
+            ->fill([
+                'status' => 0,
+                'name'   => __('admin::app.copy-of') . ' ' . $originalCartRule->name,
+            ]);
 
         $copiedCartRule->save();
 
+        foreach($copiedCartRule->channels as $channel) {
```
Comment (jitendra-webkul) [inline packages/Webkul/CartRule/src/Http/Controllers/CartRuleController.php:95]: need space after foreach

File: packages/Webkul/CartRule/src/Http/Controllers/CartRuleController.php [hunk 2]
```diff
@@ -85,10 +85,21 @@ public function copy(int $cartRuleId): View
...
+            $copiedCartRule->channels()->save($channel);
+        }
+
+        foreach($copiedCartRule->customer_groups as $group) {
```
Comment (jitendra-webkul) [inline packages/Webkul/CartRule/src/Http/Controllers/CartRuleController.php:99]: need space after foreach

File: packages/Webkul/CartRule/src/Http/Controllers/CartRuleController.php (hunk 1 above)
Comment (bosix) [reply packages/Webkul/CartRule/src/Http/Controllers/CartRuleController.php]: fixed

File: packages/Webkul/CartRule/src/Http/Controllers/CartRuleController.php (hunk 2 above)
Comment (bosix) [reply packages/Webkul/CartRule/src/Http/Controllers/CartRuleController.php]: fixed

**Pull request merged**